flake.nix

```diff
@@ -5,48 +5,97 @@
     self,
     nixpkgs,
   }: let
-    version = builtins.replaceStrings ["\n"] [""] (builtins.readFile ./version);
+    astalLib = pkgs.callPackage ./nix/lib.nix {inherit version;};
+    inherit (astalLib) getVersionFromFile mkAstalPkg;
+
+    version = getVersionFromFile ./version;
     system = "x86_64-linux";
```

Comment:
Any reason this is x86_64-linux only?

Reply:
Thats the only machine I have, but if all the dependencies can be built on other architectures, they can be added too, I just did not bother to check.